src/Site.js

```diff
@@ -667,11 +667,15 @@ Site.prototype.regenerateAffectedPages = function (filePaths) {
 /**
  * Uses heading data in built pages to generate heading and keyword information for siteData
  * Subsequently writes to siteData.json
+ * @param filePaths array of paths corresponding to files that have changed
+ * @param generateAllPages if all pages heading data should be generated
  */
-Site.prototype.generateSiteData = function () {
+Site.prototype.generateSiteData = function (filePaths = [], generateAllPages = true) {
```

Comment:
1. `generateAllPages` -> `generateForAllPages`? Since in this case we are generate site data **for** pages, not generating the pages themselves.
1. Should we swap the order of the parameters? `generateForAllPages` seems to be a much more powerful option than `filePaths`, so `generateForAllPages` can be first?